api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3293,6 +3293,92 @@ public void putObject(String bucketName, String objectName, InputStream stream,
   }
 
 
+  /**
+   * Uploads data from given stream as object to given bucket where the stream size is known.
```

Comment:
Use http://minio.github.io/minio-java/io/minio/MinioClient.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-long-java.util.Map- description as base and modify as per this method